### DISCUSSION TITLE
Are there TypeScript contents?

### DISCUSSION THREAD

**Discussion by awxiaoxian2020** [mdn/mdn-community#142] [Content]
Why don't they belong to MDN?

**Comment by Josh-Cena** [comment]
I think TypeScript is:

- Not directly essential for web development (i.e. it's not shipped to browsers)
- Proprietary (it already has its own documentation site; no need for us to reiterate on the basics)
- Hard to maintain for the content team

But if we are to have such a guide (not very likely), I can volunteer to maintain it.

**Reply by awxiaoxian2020** [reply]
My point is that we have some guides about Vue,React and etc(same as Proprietary). They are related to typescript more or less.

**Reply by Josh-Cena** [reply]
That's true, not saying it's a good idea to have those either (especially as step-by-step tutorials) :/

**Reply by teoli2003** [reply]
cc/ @Rumyra

**Reply by Rumyra** [reply]
I agree with all the points @Josh-Cena made here - TypeScript documentation is outside the scope of MDN's docs and well documented elsewhere. Moving forward we need to review the tutorials we have for React et al as they outdate quickly and maintenance for these sections can prove difficult. Having said that some overviews of web dev tooling (TypeScript, React included) might be something to consider
